ndk-build/src/readelf.rs

```diff
@@ -51,8 +51,9 @@ impl<'a> UnalignedApk<'a> {
                 };
 
                 if let Some(path) = find_library_path(search_paths, &need)? {
-                    provided.insert(path.file_name().unwrap().to_str().unwrap().to_string());
-                    artifacts.push(path);
+                    if provided.insert(path.file_name().unwrap().to_str().unwrap().to_string()) {
```

Comment:
I presume `.file_name()` gives back an equal value to `need`, which is also just a file name without path? Can we perhaps simplify this by not calling `find_library_path` if `need` is already in `provided`? OTOH I always appreciate the return value of `fn insert(...) -> bool` being used :+1: